eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -53,15 +57,31 @@ case class AuditResponse(sent: Seq[PaymentSent], received: Seq[PaymentReceived],
 
 case class TimestampQueryFilters(from: Long, to: Long)
 
+case class MutualCloseStatus(unpublished: Satoshi, unconfirmed: Satoshi, confirmed: Satoshi)
+
+case class GlobalBalance private(total: Btc, onchain: CorrectedOnchainBalance, offchain: BalanceResult, mutualClose: MutualCloseStatus)
+
+object GlobalBalance {
+
+  private def apply(total: Btc, onchain: CorrectedOnchainBalance, offchain: BalanceResult, mutualClose: MutualCloseStatus): GlobalBalance = ???
```

Comment:
The goal of this is to have the `total` attribute in the class `GlobalBalance` automatically computed as opposed to set manually (this could be bypassed by a call to `copy` though.

Why not make `total` a method? Because I want it to be automatically serialized in the json API response. We could add it manually in the json serializer but it then forces us to manually serialize all other fields, which is really ugly. Unless I'm missing something?

Reply:
See discussion https://github.com/ACINQ/eclair/pull/1737#discussion_r661306163.